### PR TITLE
 Convert Cadence Service Events to Go Type

### PR DESCRIPTION
**Pull request by danuio**
Resolves https://github.com/dapperlabs/flow-go/issues/5291.

- Adds conversion from Cadence Events to Go types for `EpochCommit` and `EpochSetup`.
- Test JSON fixtures for the Cadence events were copied from `onflow/flow-core-contracts` for testing purposes.

💡 Most of the diff is in `fixture.go` and is a JSON representation of the Cadence event taken from `flow-core-contracts`

---

### Blocking Issues
- [x] https://github.com/dapperlabs/flow-go/issues/5505

### Blocking PRs
- [x] https://github.com/onflow/flow-go/pull/678
- [x] https://github.com/onflow/flow-go/pull/661

### CODE REVIEW

**Comment by jordanschalm** [comment]
@danuio  FYI we will want to apply the canonical ordering to `EpochSetup.Participants` introduced in https://github.com/onflow/flow-go/pull/678 when we do this conversion

**Comment by jordanschalm** [comment]
`TestNetworkSetupBasic` consistently failing on CI here

**Review by arrivets — APPROVED** [review]
Nice. 
I have similar code that converts participants and clusters in the other direction, and I was wondering where to put it. Perhaps this convert package is the right place

File: model/convert/service_event.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+	}
+
+	// parse cadence types to required fields
+	setup := new(flow.EpochSetup)
+
+	// NOTE: variable names prefixed with cdc represent cadence types
+	cdcEvent := payload.(cadence.Event)
+
+	// extract simple fields
+	setup.Counter = uint64(cdcEvent.Fields[0].(cadence.UInt64))
```
Comment (turbolent) [inline model/convert/service_event.go:49]: Maybe check the field length here before accessing them, and conditionally cast, i.e. handle casting failures gracefully

File: model/convert/service_event.go [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+	// decode bytes using jsoncdc
+	payload, err := json.Decode(event.Payload)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal event payload: %w", err)
+	}
+
+	// parse cadence types to Go types
+	commit := new(flow.EpochCommit)
+	commit.Counter = uint64(payload.(cadence.Event).Fields[0].(cadence.UInt64))
```
Comment (turbolent) [inline model/convert/service_event.go:97]: Same here, maybe check the fields' length before accessing them, and conditionally cast, i.e. handle casting failures gracefully

**Review by turbolent — APPROVED** [review]
Looks good from a Go code perspective, but I don't know about the business logic, so didn't check that

**Pull request merged**